### ISSUE TITLE
[Q] PyTorch Lightning minimum sweep example

### ISSUE THREAD

**Issue by phsamuel** [wandb/wandb#5003]
Hi, I tried to play to test hyperparameter sweep with PyTorch Lightning with a minimum example (1-D regression with 1 layer hidden layer). The sweep appears to start but I keep getting error messages 

> Waiting for W&B process to finish... (failed 1). Press Control-C to abort syncing.

 I also got messages

> Run r58zdscy errored: RuntimeError('attach in the same process is not supported currently')
> wandb: ERROR Run r58zdscy errored: RuntimeError('attach in the same process is not supported currently')
> wandb: Agent Starting Run: jcqeoutz with config:
> wandb: 	lr: 0.11136099933551429
> wandb: 	n_hidden: 10

Below is my code. Anyone has any idea what I did wrong? Thanks ahead!

```
import pytorch_lightning as pl
import torch.nn.functional as F
import torch
from pytorch_lightning.loggers import WandbLogger
import wandb
from torch.utils.data import DataLoader,Dataset

class MyDataset(Dataset):
    def __init__(self, data, labels):
        self.data = data
        self.labels = labels

    def __len__(self):
        return len(self.data)

    def __getitem__(self, index):
        x = self.data[index]
        y = self.labels[index]
        return x, y

class Net(torch.nn.Module):
    def __init__(self, n_feature, n_hidden, n_output):
        super(Net, self).__init__()
        self.hidden = torch.nn.Linear(n_feature, n_hidden)
        self.predict = torch.nn.Linear(n_hidden, n_output)
    def forward(self, x):
        x= F.relu(self.hidden(x))
        x = self.predict(x)
        return x

class myLightningModule(pl.LightningModule):
    def __init__(self, config): 
        super().__init__()
        self.net = Net(n_feature=1, n_hidden=config.n_hidden, n_output=1)
        self.lr = config.lr
        self.save_hyperparameters()

    def training_step(self, batch, batch_idx):
        # training_step defines the train loop.
        x, y = batch
        y_hat = self.net(x)
        loss = F.mse_loss(y_hat, x)
        self.log('training_loss',loss,on_step=True, on_epoch=True,prog_bar=True)
        return loss

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.parameters(), lr=self.lr)
        return optimizer
    
def train_model():
    wandb.init()
    wandb_logger = WandbLogger()
        
    lightmodule = myLightningModule(wandb.config)
    trainer = pl.Trainer(accelerator='gpu', devices=1, max_epochs=20, 
        default_root_dir="./lightning-example", logger=wandb_logger)
    wandb.require(experiment="service")
    trainer.fit(model=lightmodule, train_dataloaders=my_dataloader)        

if __name__ == '__main__':
    sweep_config = {
        'method': 'random',
        'name': 'first_sweep',
        'metric': {
            'goal': 'minimize',
            'name': 'training_loss'
        },
        'parameters': {
            'n_hidden': {'values': [2,3,5,10]},
            'lr': {'max': 1.0, 'min': 0.0001}
        }
    }

    N=10000   
    x = torch.unsqueeze(torch.linspace(-1, 1, N), dim=1)                    # x data (tensor), shape=(100, 1)
    y = x.pow(2) + 0.2*torch.rand(x.size())                                  # noisy y data (tensor), shape=(100, 1)

    my_dataset = MyDataset(x,y)
    my_dataloader = DataLoader(my_dataset, batch_size=100, shuffle=True)

    sweep_id=wandb.sweep(sweep_config, project="test_sweep")
    wandb.agent(sweep_id=sweep_id, function=train_model, count=10)


```

**Comment by nate-wandb** [comment]
Hi @phsamuel, we think the issue might be related to our "service" backend. Could you try setting the env variable `WANDB_DISABLE_SERVICE=true` before you run `wandb.init()` and see if that resolves the issue? You may need to comment out `wandb.require(experiment="service")` as well. 

Let me know if that resolves the issue or if it still remains. 

Thank you,
Nate

**Comment by phsamuel** [comment]
Nate, thank you for the suggestion. Unfortunately, it seems the issue still
remains. I have the following log messages:

/home/phsamuel/p3/lib/python3.10/site-packages/pkg_resources/__init__.py:116:
PkgResourcesDeprecationWarning: 1.1build1 is an invalid version and will
not be supported in a future release
  warnings.warn(
/home/phsamuel/p3/lib/python3.10/site-packages/pkg_resources/__init__.py:116:
PkgResourcesDeprecationWarning: 0.1.43ubuntu1 is an invalid version and
will not be supported in a future release
  warnings.warn(
wandb: Agent Starting Run: 39cq76gk with config:
wandb: lr: 0.8181609041733476
wandb: n_hidden: 3
wandb: Currently logged in as: phsamuel (ann2023). Use `wandb login
--relogin` to force relogin
wandb: Tracking run with wandb version 0.13.10
wandb: Run data is saved locally in /home/phsamuel/Dropbox (Univ. of
Oklahoma)/courses/mine/deep_learning/code/wandb/run-20230222_112732-39cq76gk
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run atomic-sweep-1
wandb: ⭐️ View project at https://wandb.ai/ann2023/test_sweep
wandb: 🧹 View sweep at https://wandb.ai/ann2023/test_sweep/sweeps/kiz7a7zp
wandb: 🚀 View run at https://wandb.ai/ann2023/test_sweep/runs/39cq76gk
/home/phsamuel/p3/lib/python3.10/site-packages/pytorch_lightning/loggers/wandb.py:395:
UserWarning: There is a wandb run already in progress and newly created
instances of `WandbLogger` will reuse this run. If this is not desired,
call `wandb.finish()` before instantiating `WandbLogger`.
  rank_zero_warn(
wandb: Waiting for W&B process to finish... (failed 1). Press Control-C to
abort syncing.
wandb: 🚀 View run atomic-sweep-1 at:
https://wandb.ai/ann2023/test_sweep/runs/39cq76gk
wandb: Synced 5 W&B file(s), 0 media file(s), 0 artifact file(s) and 0
other file(s)
wandb: Find logs at: ./wandb/run-20230222_112732-39cq76gk/logs
Run 39cq76gk errored: RuntimeError('attach in the same process is not
supported currently')
wandb: ERROR Run 39cq76gk errored: RuntimeError('attach in the same process
is not supported currently')
wandb: Agent Starting Run: 3n0u5rxl with config:
wandb: lr: 0.1933642648071757
wandb: n_hidden: 5
wandb: Tracking run with wandb version 0.13.10
wandb: Run data is saved locally in /home/phsamuel/Dropbox (Univ. of
Oklahoma)/courses/mine/deep_learning/code/wandb/run-20230222_112742-3n0u5rxl
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run crisp-sweep-2
wandb: ⭐️ View project at https://wandb.ai/ann2023/test_sweep
wandb: 🧹 View sweep at https://wandb.ai/ann2023/test_sweep/sweeps/kiz7a7zp
wandb: 🚀 View run at https://wandb.ai/ann2023/test_sweep/runs/3n0u5rxl
wandb: Waiting for W&B process to finish... (failed 1). Press Control-C to
abort syncing.
wandb: - 0.002 MB of 0.008 MB uploaded (0.000 MB deduped)
wandb: \ 0.002 MB of 0.008 MB uploaded (0.000 MB deduped)
wandb: 🚀 View run crisp-sweep-2 at:
https://wandb.ai/ann2023/test_sweep/runs/3n0u5rxl
wandb: Synced 4 W&B file(s), 0 media file(s), 0 artifact file(s) and 0
other file(s)
wandb: Find logs at: ./wandb/run-20230222_112742-3n0u5rxl/logs
Run 3n0u5rxl errored: RuntimeError('attach in the same process is not
supported currently')
wandb: ERROR Run 3n0u5rxl errored: RuntimeError('attach in the same process
is not supported currently')
wandb: Agent Starting Run: cen3qu3z with config:
wandb: lr: 0.7774529226157114
wandb: n_hidden: 3
wandb: Tracking run with wandb version 0.13.10
wandb: Run data is saved locally in /home/phsamuel/Dropbox (Univ. of
Oklahoma)/courses/mine/deep_learning/code/wandb/run-20230222_112754-cen3qu3z
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run electric-sweep-3
wandb: ⭐️ View project at https://wandb.ai/ann2023/test_sweep
wandb: 🧹 View sweep at https://wandb.ai/ann2023/test_sweep/sweeps/kiz7a7zp
wandb: 🚀 View run at https://wandb.ai/ann2023/test_sweep/runs/cen3qu3z
wandb: Waiting for W&B process to finish... (failed 1). Press Control-C to
abort syncing.
wandb: - 0.002 MB of 0.002 MB uploaded (0.000 MB deduped)
wandb: \ 0.002 MB of 0.008 MB uploaded (0.000 MB deduped)
wandb: | 0.002 MB of 0.008 MB uploaded (0.000 MB deduped)
wandb: 🚀 View run electric-sweep-3 at:
https://wandb.ai/ann2023/test_sweep/runs/cen3qu3z
wandb: Synced 4 W&B file(s), 0 media file(s), 0 artifact file(s) and 0
other file(s)
wandb: Find logs at: ./wandb/run-20230222_112754-cen3qu3z/logs
Run cen3qu3z errored: RuntimeError('attach in the same process is not
supported currently')
wandb: ERROR Run cen3qu3z errored: RuntimeError('attach in the same process
is not supported currently')
Detected 3 failed runs in the first 60 seconds, killing sweep.
wandb: ERROR Detected 3 failed runs in the first 60 seconds, killing sweep.
wandb: To disable this check set WANDB_AGENT_DISABLE_FLAPPING=true
Create sweep with ID: kiz7a7zp
Sweep URL: https://wandb.ai/ann2023/test_sweep/sweeps/kiz7a7zp

===
And my current source code is as follows:

import pytorch_lightning as pl
import torch.nn.functional as F
import torch
from pytorch_lightning.loggers import WandbLogger
import wandb
from torch.utils.data import DataLoader,Dataset

class MyDataset(Dataset):
def __init__(self, data, labels):
self.data = data
self.labels = labels

def __len__(self):
return len(self.data)

def __getitem__(self, index):
x = self.data[index]
y = self.labels[index]
return x, y

class Net(torch.nn.Module):
def __init__(self, n_feature, n_hidden, n_output):
super(Net, self).__init__()
self.hidden = torch.nn.Linear(n_feature, n_hidden)
self.predict = torch.nn.Linear(n_hidden, n_output)
def forward(self, x):
x= F.relu(self.hidden(x))
x = self.predict(x)
return x

class myLightningModule(pl.LightningModule):
def __init__(self, config): #n_hidden, lr):
super().__init__()
self.net = Net(n_feature=1, n_hidden=config.n_hidden, n_output=1)
self.lr = config.lr
self.save_hyperparameters()

def training_step(self, batch, batch_idx):
# training_step defines the train loop.
x, y = batch
y_hat = self.net(x)
loss = F.mse_loss(y_hat, x)
self.log('training_loss',loss,on_step=True, on_epoch=True,prog_bar=True)
return loss

def configure_optimizers(self):
optimizer = torch.optim.SGD(self.parameters(), lr=self.lr)
return optimizer
def train_model():
WANDB_DISABLE_SERVICE=True
wandb.init()
wandb_logger = WandbLogger()
lightmodule = myLightningModule(wandb.config)
trainer = pl.Trainer(accelerator='gpu', devices=1, max_epochs=20,
default_root_dir="./lightning-example", logger=wandb_logger)
# wandb.require(experiment="service")
trainer.fit(model=lightmodule, train_dataloaders=my_dataloader)

if __name__ == '__main__':
sweep_config = {
'method': 'random',
'name': 'first_sweep',
'metric': {
'goal': 'minimize',
'name': 'training_loss'
},
'parameters': {
'n_hidden': {'values': [2,3,5,10]},
'lr': {'max': 1.0, 'min': 0.0001}
}
}

N=10000
x = torch.unsqueeze(torch.linspace(-1, 1, N), dim=1) # x data (tensor),
shape=(100, 1)
y = x.pow(2) + 0.2*torch.rand(x.size()) # noisy y data (tensor),
shape=(100, 1)

my_dataset = MyDataset(x,y)
my_dataloader = DataLoader(my_dataset, batch_size=100, shuffle=True)

sweep_id=wandb.sweep(sweep_config, project="test_sweep")
wandb.agent(sweep_id=sweep_id, function=train_model, count=10)

**Comment by phsamuel** [comment]
The problem is resolved. It appears that there is a conflict with Lightning's save_hyperparameters(). If I commented that line, it seems to be fine. But I have made lots of changes and in case someone is interested, the following seems to work. 

```
import pytorch_lightning as pl
import torch.nn.functional as F
import torch
from pytorch_lightning.loggers import WandbLogger
import wandb
from torch.utils.data import DataLoader,Dataset
import os

import torch
from torch.utils.data import Dataset


class MyDataset(Dataset):
    def __init__(self, data, labels):
        self.data = data
        self.labels = labels

    def __len__(self):
        return len(self.data)

    def __getitem__(self, index):
        x = self.data[index]
        y = self.labels[index]
        return x, y
    
class MyDataModule(pl.LightningDataModule):
    def __init__(self, config):
        super().__init__()
        N=10000   
        x = torch.unsqueeze(torch.linspace(-1, 1, N), dim=1)                    # x data (tensor), shape=(100, 1)
        y = x.pow(2) + config.noise*torch.rand(x.size())                                  # noisy y data (tensor), shape=(100, 1)
        self.my_dataset = MyDataset(x,y)
        print(config.noise)

    def train_dataloader(self):
        return DataLoader(self.my_dataset,batch_size=100,shuffle=True)

    def val_dataloader(self):
        return DataLoader(self.my_dataset,batch_size=100,shuffle=False)



class Net(torch.nn.Module):
    def __init__(self, n_feature, n_hidden, n_output):
        super(Net, self).__init__()
        self.hidden = torch.nn.Linear(n_feature, n_hidden)
        self.predict = torch.nn.Linear(n_hidden, n_output)
    def forward(self, x):
        x= F.relu(self.hidden(x))
        x = self.predict(x)
        return x

class MyLightningModule(pl.LightningModule):
    def __init__(self, config): #n_hidden, lr):
        super().__init__()
        self.net = Net(n_feature=1, n_hidden=config.n_hidden, n_output=1)
        self.lr = config.lr
#         self.save_hyperparameters()

    def training_step(self, batch, batch_idx):
        # training_step defines the train loop.
        x, y = batch
        y_hat = self.net(x)
        loss = F.mse_loss(y_hat, x)
        self.log('training_loss',loss,on_step=True, on_epoch=True,prog_bar=True)
        return loss
    
    def validation_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self.net(x)
        loss = F.mse_loss(y_hat, x)
        self.log('validation_loss',loss)
        # return {'val_loss': loss}
        return loss    

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.parameters(), lr=self.lr)
        return optimizer
    
def train_model():
#     os.environ["WANDB_START_METHOD"] = "thread"'
    wandb.init(project="sweep")
    config=wandb.config
    wandb_logger = WandbLogger()
    data = MyDataModule(config)
    module = MyLightningModule(config)

    wandb_logger.watch(module.net)
    
    trainer = pl.Trainer(accelerator='gpu', devices=1, max_epochs=10, 
        default_root_dir="./lightning-example", logger=wandb_logger)
#     wandb.require(experiment="service")
    trainer.fit(module, data)        

    
    
if __name__ == '__main__':
    sweep_config = {
        'method': 'random',
        'name': 'first_sweep',
        'metric': {
            'goal': 'minimize',
            'name': 'validation_loss'
        },
        'parameters': {
            'n_hidden': {'values': [2,3,5,10]},
            'lr': {'max': 1.0, 'min': 0.0001},
            'noise': {'max': 1.0, 'min': 0.}
        }
    }

    sweep_id=wandb.sweep(sweep_config, project="test_sweep")
    wandb.agent(sweep_id=sweep_id, function=train_model, count=5)
```

**Comment by nate-wandb** [comment]
Glad to hear this is resolved and thank you posting the solution!

**Comment by Kin-Zhang** [comment]
But I think wandb sweep problem still exists in DDP running, with error mentioned here: https://github.com/wandb/wandb/issues/6449
